trino/client.py

```diff
@@ -594,7 +594,8 @@ class TrinoResult(object):
 
     def __init__(self, query, rows=None, experimental_python_types: bool = False):
         self._query = query
-        self._rows = rows or []
+        # Initial rows from the first POST request
```

Comment:
There's no guarantee initial rows will be present

Reply:
2 points here:

- It seems that there is a `TrinoResult` instantiation in  https://github.com/trinodb/trino-python-client/blob/aee6064c53536da6a73d0f2fa387eb2dfb222600/trino/client.py#L726. I'm not sure why that would be useful, as before fetching a query should be executed which will override the `_result` field in `TrinoQuery`.
- Trino's `QueryResults` can indeed expose a `null` data field, potentially on each fetch response, but that's handled in https://github.com/trinodb/trino-python-client/blob/aee6064c53536da6a73d0f2fa387eb2dfb222600/trino/client.py#L568